rust/kernel/module_param.rs

```diff
@@ -69,11 +69,9 @@ pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
         val: *const crate::c_types::c_char,
         param: *const crate::bindings::kernel_param,
     ) -> crate::c_types::c_int {
-        let arg = if val.is_null() {
-            None
-        } else {
-            Some(unsafe { CStr::from_char_ptr(val).as_bytes() })
-        };
+        let arg = val
+            .is_null()
+            .then_some( unsafe { CStr::from_char_ptr(val).as_bytes() });
```

Comment:
I think these all have the condition flipped. In addition the `CStr::from_char_ptr` call must not happen in case of `.is_null()` being true.

Reply:
oh, very true. I don't know how I missed that.